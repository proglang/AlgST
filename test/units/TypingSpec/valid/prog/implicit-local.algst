calc : Int -> Int

check : Int
check =
  let {number} = 42 in
  calc (implicit[Int] + 10)

synth : Int
synth =
  let {number} = 42 in
  let r = calc (implicit[Int] + 10) in
  r
