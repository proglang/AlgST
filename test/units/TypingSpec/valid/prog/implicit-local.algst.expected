calc#M.2
|
`- SignatureDecl
   |
   `- type
      |
      `- Type.Arrow U
         |
         +- Type.Type
         |  |
         |  `- TypeRef
         |     |
         |     `- Int
         |
         `- Type.Type
            |
            `- TypeRef
               |
               `- Int

check#M.0
|
`- ValueDecl
   |
   +- type
   |  |
   |  `- Type.Type
   |     |
   |     `- TypeRef
   |        |
   |        `- Int
   |
   +- params
   |
   `- definition
      |
      `- Exp.Exp
         |
         `- TcExpX.ValueCase
            |
            +- Exp.Lit
            |  |
            |  `- Lit.Int 42
            |
            `- number#M.3
               |
               `- Exp.App
                  |
                  +- Exp.Var calc#M.2
                  |
                  `- Exp.App
                     |
                     +- Exp.App
                     |  |
                     |  +- Exp.Var (+)#Builtin.19
                     |  |
                     |  `- Exp.App
                     |     |
                     |     +- Exp.TypeApp
                     |     |  |
                     |     |  +- Exp.Var implicit#Builtin.32
                     |     |  |
                     |     |  `- Type.Type
                     |     |     |
                     |     |     `- TypeRef
                     |     |        |
                     |     |        `- Int
                     |     |
                     |     `- Exp.Var number#M.3
                     |
                     `- Exp.Lit
                        |
                        `- Lit.Int 10

synth#M.1
|
`- ValueDecl
   |
   +- type
   |  |
   |  `- Type.Type
   |     |
   |     `- TypeRef
   |        |
   |        `- Int
   |
   +- params
   |
   `- definition
      |
      `- Exp.Exp
         |
         `- TcExpX.ValueCase
            |
            +- Exp.Lit
            |  |
            |  `- Lit.Int 42
            |
            `- number#M.4
               |
               `- Exp.Exp
                  |
                  `- TcExpX.ValueCase
                     |
                     +- Exp.App
                     |  |
                     |  +- Exp.Var calc#M.2
                     |  |
                     |  `- Exp.App
                     |     |
                     |     +- Exp.App
                     |     |  |
                     |     |  +- Exp.Var (+)#Builtin.19
                     |     |  |
                     |     |  `- Exp.App
                     |     |     |
                     |     |     +- Exp.TypeApp
                     |     |     |  |
                     |     |     |  +- Exp.Var implicit#Builtin.32
                     |     |     |  |
                     |     |     |  `- Type.Type
                     |     |     |     |
                     |     |     |     `- TypeRef
                     |     |     |        |
                     |     |     |        `- Int
                     |     |     |
                     |     |     `- Exp.Var number#M.4
                     |     |
                     |     `- Exp.Lit
                     |        |
                     |        `- Lit.Int 10
                     |
                     `- r#M.5
                        |
                        `- Exp.Var r#M.5

